backport_packages/refactor_backport_packages.py

```diff
@@ -0,0 +1,369 @@
+        :param names: Name of the file
+        :return:
+        """
+        ignored_names = []
+        if any([src.endswith(os.path.sep + class_type) for class_type in CLASS_TYPES]):
+            ignored_names = [name for name in names
+                             if is_bigquery_non_dts_module(module_name=name)]
+        if src.endswith(os.path.sep + "example_dags"):
+            for file_name in names:
+                file_path = src + os.path.sep + file_name
+                with open(file_path, "rt") as file:
+                    text = file.read()
+                if any([f"airflow.providers.google.cloud.{class_type}.bigquery" in text
+                        for class_type in CLASS_TYPES]) or "_to_bigquery" in text:
+                    print(f"Ignoring {file_path}")
+                    ignored_names.append(file_name)
+        return ignored_names
+
+    def ignore_kubernetes_files(src: str, names: List[str]) -> List[str]:
+        ignored_names = []
+        if src.endswith(os.path.sep + "example_dags"):
+            for file_name in names:
+                if "example_kubernetes" in file_name:
+                    ignored_names.append(file_name)
+        return ignored_names
+
+    def ignore_some_files(src: str, names: List[str]) -> List[str]:
+        ignored_list = ignore_bigquery_files(src=src, names=names)
+        ignored_list.extend(ignore_kubernetes_files(src=src, names=names))
+        return ignored_list
+
+    rm_build_dir()
+    package_providers_dir = get_target_providers_folder()
+    if os.path.isdir(package_providers_dir):
+        rmtree(package_providers_dir)
+    copytree(get_source_providers_folder(), get_target_providers_folder(), ignore=ignore_some_files)
+
+
+class RefactorBackportPackages:
+    """
+    Refactors the code of providers, so that it works in 1.10.
+
+    """
+
+    def __init__(self):
+        self.qry = Query()
+
+    def remove_class(self, class_name) -> None:
+        # noinspection PyUnusedLocal
+        def _remover(node: LN, capture: Capture, filename: Filename) -> None:
+            if node.type not in (300, 311):  # remove only definition
+                node.remove()
+
+        self.qry.select_class(class_name).modify(_remover)
+
+    def rename_deprecated_modules(self):
+        changes = [
+            ("airflow.operators.bash", "airflow.operators.bash_operator"),
+            ("airflow.operators.python", "airflow.operators.python_operator"),
+            ("airflow.utils.session", "airflow.utils.db"),
+            (
+                "airflow.providers.cncf.kubernetes.operators.kubernetes_pod",
+                "airflow.contrib.operators.kubernetes_pod_operator"
+            ),
+        ]
+        for new, old in changes:
+            self.qry.select_module(new).rename(old)
+
+    def add_provide_context_to_python_operators(self):
+        # noinspection PyUnusedLocal
+        def add_provide_context_to_python_operator(node: LN, capture: Capture, filename: Filename) -> None:
```

Comment:
why do we need to add this one back?

This means the API of the operator will be different between backport and eventual 2.0.

If they want the old interface, user's can stick with the non backported op

Reply:
The PythonOperaotra interface has changed in Airflow 2.0. In airflow 1.10, you had to provide additional options for the context to be passed. In Airflow 2.0, this happens automatically. For the example to work properly on Airflow 1.10, we need to add this parameter.
https://github.com/apache/airflow/pull/5990

Reply:
Yes. @mik-laj explained it well. We have to do it this way,

Reply:
Yes, I'm aware of the change to PythonOperator.

My point is: why do we need to reverse that change for backported operators? Isn't the point of backporting to make the new interface available?

Reply:
Good question. And we have good answers I think:

First of all: this change (see #8991 for examples) only applies to example days that use the Python operators.

Secondly: We are not backporting (and we've never planned to backport)  Python operators - they are part of the "core" operators, not the providers. The discussion we had back then was that some of the core operators (Python/Bash/Branch/Dummy/LatestOnly/BranchPython, ShortCircuit/PythonVirtualEnv - I believe that's a complete list but I could miss one or two) are really part of the Airflow Core and should stay there. The idea is that without those operators Airflow is fairly useless so they were supposed to be delivered with Airflow itself. They were moved to another location to make the naming consistent (but with deprecation redirection from old locations). But if any example uses any of them (quite a few) we had to refactor the examples to make them work with the 1.10.* version of those operators. That was a similar thing with tags (which we removed from examples so that the examples are applicable to pre-1.10.* version of the providers as well).

So the decision was that they are going to stay in core and that their version is linked to Airflow version they are from.

Obviously we could revert that decision  - but if we do so - we should add another provider ("core" ?). And we probably need to re-open discussion about that - because we all agreed on that before when we presented list of operators and explained that those will stay in core. 

I do not think it's a blocker for this backport package release I believe) - It can be done easily afterwards as a separate release. Let me know if you think otherwise.  

Feel free to open discussion on that @ash on devlist if you think it makes sense to release some of those as separate "provider".